api/v1alpha1/gateway_types.go

```diff
@@ -148,15 +178,41 @@ const (
 // - gcp: https://cloud.google.com/load-balancing/docs/use-ssl-policies#creating_an_ssl_policy_with_a_custom_profile
 // - aws: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html#describe-ssl-policies
 // - azure: https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-bindings#enforce-tls-1112
-type ListenerTLS struct {
-	// Certificates is a list of certificates containing resources
-	// that are bound to the listener.
+type TLSConfig struct {
+	// Certificates is a reference to one or more Kubernetes objects each containing
+	// an identity certificate that is bound to a listener. The hostname in a TLS
+	// SNI client hello message is used for certificate matching and route hostname
+	// selection.
 	//
 	// If apiGroup and kind are empty, will default to Kubernetes Secrets resources.
 	//
 	// Support: Core (Kubernetes Secrets)
 	// Support: Implementation-specific (Other resource types)
-	Certificates []core.TypedLocalObjectReference `json:"certificates,omitempty"`
+	//
+	// +optional
+	ListenerCertificates []core.TypedLocalObjectReference `json:"listenerCertificates,omitempty"`
```

Comment:
I think this relates back to my comment on `Listener.TLS`. A collection of listener certificates makes sense when you are modeling a single port that can terminate different TLS names. However, further down, `TLSConfig` is used in `HTTPRouteAction`, and in that context it's not clear to me whether this is a pinned set of certificates that the client will verify and accept, or simply not used. Attaching the TLS server configuration to the `HTTPHostRoute` could help to resolve this ambiguity.

Reply:
My current understand is that the certs associated with the Gateway here are for downstream connection and the TLSConfig in `HostRouteAction` is for the upstream hop.